packages/medusa/src/services/store.ts

```diff
@@ -0,0 +1,290 @@
+import { MedusaError } from "medusa-core-utils"
+import { BaseService } from "medusa-interfaces"
+import { currencies, Currency } from "../utils/currencies"
+import { EntityManager } from "typeorm"
+import { StoreRepository } from "../repositories/store"
+import { CurrencyRepository } from "../repositories/currency"
+import EventBusService from "./event-bus"
+import { Store } from "../models"
+import { AdminPostStoreReq } from "../api/routes/admin/store"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  storeRepository: typeof StoreRepository
+  currencyRepository: typeof CurrencyRepository
+  eventBusService: EventBusService
+}
+
+/**
+ * Provides layer to manipulate store settings.
+ * @extends BaseService
+ */
+class StoreService extends BaseService {
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager
+  protected readonly storeRepository_: typeof StoreRepository
+  protected readonly currencyRepository_: typeof CurrencyRepository
+  protected readonly eventBus_: EventBusService
+
+  constructor({
+    manager,
+    storeRepository,
+    currencyRepository,
+    eventBusService,
+  }: InjectedDependencies) {
+    super()
+
+    this.manager_ = manager
+    this.storeRepository_ = storeRepository
+    this.currencyRepository_ = currencyRepository
+    this.eventBus_ = eventBusService
+  }
+
+  withTransaction(transactionManager): this | StoreService {
+    if (!transactionManager) {
+      return this
+    }
+
+    const cloned = new StoreService({
+      manager: transactionManager,
+      storeRepository: this.storeRepository_,
+      currencyRepository: this.currencyRepository_,
+      eventBusService: this.eventBus_,
+    })
+
+    cloned.transactionManager_ = transactionManager
+
+    return cloned
+  }
+
+  /**
+   * Creates a store if it doesn't already exist.
+   * @return The store.
+   */
+  async create(): Promise<Store> {
+    return await this.atomicPhase_(
+      async (transactionManager: EntityManager) => {
+        const storeRepository = transactionManager.getCustomRepository(
+          this.storeRepository_
+        )
+        const currencyRepository = transactionManager.getCustomRepository(
+          this.currencyRepository_
+        )
+
+        let store = await this.retrieve()
+
+        if (!store) {
+          const newStore = await storeRepository.create()
+          // Add default currency (USD) to store currencies
+          const usd = await currencyRepository.findOne({
+            code: "usd",
+          })
+
+          if (usd) {
+            newStore.currencies = [usd]
+          }
+
+          store = await storeRepository.save(newStore)
+        }
+
+        return store
+      }
+    )
+  }
+
+  /**
+   * Retrieve the store settings. There is always a maximum of one store.
+   * @param relations - relations to fetch with store
+   * @return the store
+   */
+  async retrieve(relations: string[] = []): Promise<Store> {
```

Comment:
Now that we are refactoring the StoreService, we should align this method with the general approach by adding a config object. See [CustomerService for reference](https://github.com/medusajs/medusa/blob/eee780f110b70fcd4e8f982684b5f4ecd4128def/packages/medusa/src/services/customer.js#L238). 

Change should be reflected in the controller.

Reply:
I know, that the logic in this method is slightly different from other retrieve's, but I still think it's reasonable to do to improve DX.

cc @srindom